### PR TITLE
Add the option to open more than one specific pages on startup

### PR DESCRIPTION
**Pull request by ChihweiLHBird**
Attempted to solve issue: #1316

### CODE REVIEW

**Comment by Jaiganeshkumaran** [comment]
@ChihweiLHBird Could you share a screenshot of the page?

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird Could you share a screenshot of the page?

image

**Comment by ChihweiLHBird** [comment]
> Change the content from 'Open a specific page' to 'Open a specific page or pages'. Also, update the resource file.

The content has been updated. How to update the resource file? Does Visual Studio do it automatically?

**Comment by Jaiganeshkumaran** [comment]
> > Change the content from 'Open a specific page' to 'Open a specific page or pages'. Also, update the resource file.
> 
> The content has been updated. How to update the resource file? Does Visual Studio do it automatically?
Under the strings folder, edit the resources.resw file in the en-US folder and then build the project to apply to resource files for other languages.

**Comment by ChihweiLHBird** [comment]
> > > Change the content from 'Open a specific page' to 'Open a specific page or pages'. Also, update the resource file.
> > 
> > 
> > The content has been updated. How to update the resource file? Does Visual Studio do it automatically?
> 
> Under the strings folder, edit the resources.resw file in the en-US folder and then build the project to apply to resource files for other languages.

My Visual Studio does not apply it to other languages when building the project...

**Comment by Jaiganeshkumaran** [comment]
> My Visual Studio does not apply it to other languages when building the project...

You need to install the Multilingual App Toolkit Extension.

**Comment by Jaiganeshkumaran** [comment]
Instead of 'delete page' and 'edit page', it would be better if it says 'remove page' and 'change page location'.

**Comment by ChihweiLHBird** [comment]
> Instead of 'delete page' and 'edit page', it would be better if it says 'remove page' and 'change page location'.

How about "Change Page"?

**Comment by Jaiganeshkumaran** [comment]
> > Instead of 'delete page' and 'edit page', it would be better if it says 'remove page' and 'change page location'.
> 
> How about "Change Page"?

First of all, these are not pages but locations. So change page location should be more appropriate.

**Comment by ChihweiLHBird** [comment]
Okay, and it is updated.

**Comment by Jaiganeshkumaran** [comment]
Make the edit button before the change buttons. It looks weird to me,

**Comment by ChihweiLHBird** [comment]
How is it now?

**Comment by Jaiganeshkumaran** [comment]
> How is it now?

Great!

**Comment by ChihweiLHBird** [comment]
I am going to the bed. Let me know if you find some other issues. I will fix them tomorrow. (●'◡'●)

**Comment by ChihweiLHBird** [comment]
> Great work! This should also help with restoring opened locations in the future.

Thank you!

File: Files/Views/InstanceTabsView.xaml.cs [hunk 1]
```diff
@@ -99,7 +98,11 @@ protected override void OnNavigatedTo(NavigationEventArgs eventArgs)
             {
                 try
                 {
-                    AddNewTab(typeof(ModernShellPage), App.AppSettings.OpenASpecificPageOnStartupPath);
+                    if (App.AppSettings.PagesOnStartupList != null)
```
Comment (tsvietOK) [inline Files/Views/InstanceTabsView.xaml.cs:101]: Can you add braces for all if's and foreach's.

File: Files/Views/SettingsPages/OnStartup.xaml [hunk 2]
```diff
@@ -50,31 +49,16 @@
...
-
-                    <Button
-                        x:Name="btnOpenASpecificPageOnStartupPath"
-                        Height="33"
-                        Click="btnOpenASpecificPageOnStartupPath_Click"
-                        Content="&#xE838;"
-                        CornerRadius="0,2,2,0"
-                        FontFamily="Segoe MDL2 Assets"
-                        Style="{ThemeResource AccentButtonStyle}" />
+                <StackPanel x:Name="PagesPanel" Width="550" HorizontalAlignment="Left" Visibility="{x:Bind OpenSpecificPage.IsChecked, Mode=OneWay}">
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/OnStartup.xaml:57]: Each property should be at the new line.

File: Files/Views/SettingsPages/OnStartup.xaml.cs [hunk 3]
```diff
@@ -3,19 +3,28 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Windows.UI.Xaml;
 
 namespace Files.SettingsPages
 {
     public sealed partial class OnStartup : Page
     {
         public SettingsViewModel AppSettings => App.AppSettings;
+        //public List<string> SpecificPagesOnStartup { get; set; }
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/OnStartup.xaml.cs:16]: Is it needed?

File: Files/Views/SettingsPages/OnStartup.xaml.cs [hunk 4]
```diff
@@ -3,19 +3,28 @@
...
 
         public OnStartup()
         {
             InitializeComponent();
+
+            if (AppSettings.OpenASpecificPageOnStartup && AppSettings.PagesOnStartupList?.Length > 0)
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/OnStartup.xaml.cs:22]: Add braces.

File: Files/Views/SettingsPages/OnStartup.xaml.cs [hunk 5]
```diff
@@ -24,9 +33,41 @@ private async void btnOpenASpecificPageOnStartupPath_Click(object sender, Window
 
             if (folder != null)
             {
-                // Application now has read/write access to the picked file
-                AppSettings.OpenASpecificPageOnStartupPath = folder.Path;
+                if (AppSettings.PagesOnStartupList != null)
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/OnStartup.xaml.cs:36]: Add braces

File: Files/Views/SettingsPages/OnStartup.xaml.cs [hunk 6]
```diff
@@ -24,9 +33,41 @@ private async void btnOpenASpecificPageOnStartupPath_Click(object sender, Window
...
+                    AppSettings.PagesOnStartupList[i] = new_path;
+                    break;
+                }
+        }
+
+        public void RemovePage(SpecificPageOnStartup pageItem, string path)
+        {
+            PagesPanel.Children.Remove(pageItem);
+            string [] newPages = AppSettings.PagesOnStartupList.Where(s => s != path).ToArray();
+            if (newPages.Length > 0)
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/OnStartup.xaml.cs:67]: Add braces

File: Files/UserControls/SpecificPageOnStartup.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+using Windows.Storage.Pickers;
+using Files.View_Models;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace Files.UserControls
+{
+    public sealed partial class SpecificPageOnStartup : UserControl
+    {
+        //public SettingsViewModel AppSettings => App.AppSettings;
```
Comment (tsvietOK) [inline Files/UserControls/SpecificPageOnStartup.xaml.cs:25]: Is it needed?

File: Files/UserControls/SpecificPageOnStartup.xaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                            <FontIcon Glyph="&#xE70F;" />
+                        </MenuFlyoutItem.Icon>
+                    </MenuFlyoutItem>
+                    <MenuFlyoutItem
+                        x:Uid="RemovePage"
+                        Text="Remove Page"
+                        Click="RemovePage_Click">
+                        <MenuFlyoutItem.Icon>
+                            <FontIcon Glyph="&#xE74D;" />
+                        </MenuFlyoutItem.Icon>
```
Comment (XPoppyX) [inline Files/UserControls/SpecificPageOnStartup.xaml:47]: fluent icons should be used

Comment (ChihweiLHBird) [reply Files/UserControls/SpecificPageOnStartup.xaml:47]: How to use fluent icons? I used Segoe MDL2 icons from https://docs.microsoft.com/en-us/windows/uwp/design/style/segoe-ui-symbol-font, and I think that's part of fluent design?

Comment (XPoppyX) [reply Files/UserControls/SpecificPageOnStartup.xaml:47]: all icons have been changed and mdl2 icons are no longer used. See: #1374 


just change them to this:


Anotación 2020-07-14 211257

**Comment by ChihweiLHBird** [comment]
> image
> Can you decrease size of Pages text and align(left) path bars with Pages text.

All done, thank you for reviewing.

File: Files/UserControls/SpecificPageOnStartup.xaml (hunk 8 above)
Comment (ChihweiLHBird) [reply Files/UserControls/SpecificPageOnStartup.xaml]: Thank you for providing this. Where can I find a list of fluent icons on Microsoft's website?

Comment (XPoppyX) [reply Files/UserControls/SpecificPageOnStartup.xaml]: here: https://github.com/microsoft/fluentui-system-icons/blob/master/icons.md

**Comment by tsvietOK** [comment]
image
And it is better to change Add page text to "Plus" button and make it little bit smaller.

**Comment by ChihweiLHBird** [comment]
> image
> Still not aligned to the left

Sorry about that. It should be okay now.
image

**Comment by ChihweiLHBird** [comment]
> image
> And it is better to change Add page text to "Plus" button and make it little bit smaller.

What icon is it?

**Comment by tsvietOK** [comment]
> > image
> > And it is better to change Add page text to "Plus" button and make it little bit smaller.
> 
> What icon is it?

`Glyph="&#xE903;"`

**Comment by tsvietOK** [comment]
@ChihweiLHBird 
image

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird
> image

Would you like left margin (of the add button) to be 10 or a higher value?

File: Files/Views/InstanceTabsView.xaml.cs [hunk 9]
```diff
@@ -99,7 +98,18 @@ protected override void OnNavigatedTo(NavigationEventArgs eventArgs)
...
+                    {
+                        foreach (string path in App.AppSettings.PagesOnStartupList)
+                        {
+                            AddNewTab(typeof(ModernShellPage), path);
+                        }
+                    }
+
```
Comment (tsvietOK) [inline Files/Views/InstanceTabsView.xaml.cs:108]: Remove new line

**Comment by duke7553** [comment]
@ChihweiLHBird Look's great! Here's some final feedback:

- Add text back to button
- Fix alignment with pages text with controls below

**Comment by XPoppyX** [comment]
It looks weird being so close to "pages:"

> image
> And it is better to change Add page text to "Plus" button and make it little bit smaller.

**Comment by tsvietOK** [comment]
@XPoppyX It is just an example, we could leave it at the right side.
@duke7553 I think it is better when button text(Add pages) is not repeating text before button(Pages), but you have the final say.

**Comment by ChihweiLHBird** [comment]
I would prefer the one come with Edge.
image

File: Files/Views/SettingsPages/OnStartup.xaml [hunk 10]
```diff
@@ -62,12 +62,12 @@
                     <StackPanel 
                         Orientation="Horizontal" 
                         Margin="0,10,0,20">
-                        <TextBlock Text="Pages: " FontSize="18" />
+                        <TextBlock Text="Pages: " FontSize="18" HorizontalAlignment="Stretch" VerticalAlignment="Center" />
                         <Button 
                             x:Name="btnAddPage" 
-                            Content="Add Page" 
+                            Content="Add a new page"
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/OnStartup.xaml:68]: Can you add localization for the new strings?

**Comment by ChihweiLHBird** [comment]
image

Everyone, how is this?

File: Files/Views/SettingsPages/OnStartup.xaml (hunk 10 above)
Comment (ChihweiLHBird) [reply Files/Views/SettingsPages/OnStartup.xaml:68]: Okay

Comment (ChihweiLHBird) [reply Files/Views/SettingsPages/OnStartup.xaml:74]: It has been done.

File: Files/MultilingualResources/Files.de-DE.xlf [hunk 11]
```diff
@@ -977,6 +977,22 @@
...
+        </trans-unit>
+        <trans-unit id="SettingsOnStartupOpenSpecificPagesUserControlChangePage.Text" translate="yes" xml:space="preserve">
+          <source>Change Page Location</source>
+          <target state="new">Change Page Location</target>
+        </trans-unit>
+        <trans-unit id="SettingsOnStartupPages.Text" translate="yes" xml:space="preserve">
+          <source>Pages:</source>
+          <target state="new">Pages:</target>
+        </trans-unit>
+        <trans-unit id="String1SettingsOnStartupOpenSpecificPagesUserControlRemovePage" translate="yes" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.de-DE.xlf:992]:
```suggestion
        <trans-unit id="SettingsOnStartupOpenSpecificPagesUserControlRemovePage" translate="yes" xml:space="preserve">
```

**Pull request merged**